src/Microsoft.Framework.Runtime.Roslyn/RoslynCompiler.cs

```diff
@@ -186,6 +189,29 @@ public class RoslynCompiler
             return compilationContext;
         }
 
+        private CSharpCompilation ApplyStrongNameSettings(CSharpCompilation compilation)
+        {
+#if DNX451
+            // This is temporary, eventually we'll want a project.json feature for this
+            var keyFile = Environment.GetEnvironmentVariable(EnvironmentNames.BuildKeyFile);
+            if(!string.IsNullOrEmpty(keyFile))
+            {
+                var delaySignString = Environment.GetEnvironmentVariable(EnvironmentNames.BuildDelaySign);
+                var delaySign = !string.IsNullOrEmpty(delaySignString) && (
+                    string.Equals(delaySignString, "true", StringComparison.OrdinalIgnoreCase) ||
+                    string.Equals(delaySignString, "1", StringComparison.OrdinalIgnoreCase));
+
+                var strongNameProvider = new DesktopStrongNameProvider();
+                var newOptions = compilation.Options
+                    .WithStrongNameProvider(strongNameProvider)
+                    .WithCryptoKeyFile(keyFile)
+                    .WithDelaySign(!string.IsNullOrEmpty(delaySignString) && string.Equals("true", delaySignString, StringComparison.OrdinalIgnoreCase));
+                return compilation.WithOptions(newOptions);
+            }
+#endif
```

Comment:
`#elseif throw <something>` or add a diagnostic or something? Or maybe put the `#ifdef` inside the main `if()`? I could imagine debugging this would be painful otherwise...